crates/hir-ty/src/tests/regression.rs

```diff
@@ -1526,6 +1526,34 @@ unsafe impl Storage for InlineStorage {
     );
 }
 
+#[test]
+fn gat_crash_3() {
+    // FIXME: This test currently crashes rust analyzer in a debug build but not in a
+    // release build (i.e. for the user). With the assumption that tests will always be run
+    // in debug mode, we catch the unwind and expect that it panicked. See the
+    // [`crate::utils::generics`] function for more information.
+    cov_mark::check!(ignore_gats);
+    std::panic::catch_unwind(|| {
```

Comment:
@flodiebold So, to make the test succeed I just catch the unwind here right now, which is pretty much the same as `#[shoud_panic]`. Or can we merge without passing tests?